modules/tensor_mechanics/src/materials/ComputeLagrangianStrain.C

```diff
@@ -0,0 +1,213 @@
+#include "ComputeLagrangianStrain.h"
+
+registerMooseObject("TensorMechanicsApp", ComputeLagrangianStrain);
+
+InputParameters
+ComputeLagrangianStrain::validParams()
+{
+  InputParameters params = Material::validParams();
+
+  params.addRequiredCoupledVar("displacements", "Displacement variables");
+  params.addParam<bool>(
+      "large_kinematics", false, "Use large displacement kinematics in the kernel.");
+  params.addParam<std::vector<MaterialPropertyName>>("eigenstrain_names",
+                                                     "List of eigenstrains to account for");
+  params.addCoupledVar("macro_gradient", "Optional scalar field with the macro gradient");
+
+  // We rely on this *not* having use_displaced mesh on
+  params.suppressParameter<bool>("use_displaced_mesh");
+
+  return params;
+}
+
+ComputeLagrangianStrain::ComputeLagrangianStrain(const InputParameters & parameters)
+  : DerivativeMaterialInterface<Material>(parameters),
+    _ndisp(coupledComponents("displacements")),
+    _disp(3),
+    _grad_disp(3),
+    _ld(getParam<bool>("large_kinematics")),
+    _eigenstrain_names(getParam<std::vector<MaterialPropertyName>>("eigenstrain_names")),
+    _eigenstrains(_eigenstrain_names.size()),
+    _eigenstrains_old(_eigenstrain_names.size()),
+    _total_strain(declareProperty<RankTwoTensor>("total_strain")),
```

Comment:
Same suggestion as earlier: please use prepend `_base_name` to material properties whenever possible in tensor mechanics.